ts/Series/PointAndFigure/PointAndFigureSeries.ts

```diff
@@ -0,0 +1,361 @@
+                        pushNewPoint(close, upTrend, lastPoint);
+                    }
+                }
+
+                if (!upTrend) {
+                    // Add point going DOWN
+                    if (lastPoint - close >= calculatedBoxSize) {
+                        pushNewPoint(close, upTrend, lastPoint);
+                    }
+
+                    if (close - lastPoint >= reversal) { // Handle reversal
+                        upTrend = true;
+
+                        pushPointGroup(x, [], true);
+                        pushNewPoint(close, upTrend, lastPoint);
+                    }
+                }
+            });
+        }
+
+        // Process the pnfDataGroups to HC series format
+        const finalData: {x: number, y: number, upTrend: boolean}[] = [];
+
+        series.processedXData.length = 0;
+        series.processedYData.length = 0;
+
+        pnfDataGroups.forEach((point): void => {
+            const x = point.x,
+                upTrend = point.upTrend;
+
+            point.y.forEach((y): void => {
+                series.processedXData.push(x);
+                series.processedYData.push(y as any);
+                finalData.push({
+                    x,
+                    y,
+                    upTrend
+                });
+            });
+        });
+
+        series.pnfDataGroups = pnfDataGroups;
+        series.processedData = finalData;
+    }
+
+    public markerAttribs(
+        point: Point
+    ): SVGAttributes {
+        const series = this,
+            options = series.options,
+            attribs: SVGAttributes = {},
+            pos = point.pos();
+
+        attribs.width = series.markerWidth;
+        attribs.height = series.markerHeight;
+        if (pos && attribs.width && attribs.height) {
+            attribs.x = pos[0] - Math.round(attribs.width) / 2;
+            attribs.y = pos[1] - Math.round(attribs.height) / 2;
+        }
+        if (options.crisp && attribs.x) {
+        // Math.floor for #1843:
+            attribs.x = Math.floor(attribs.x);
+        }
+
+        return attribs;
+    }
+
+    public translate(): void {
+        const metrics = this.getColumnMetrics(),
+            calculatedBoxSize = this.calculatedBoxSize;
+
+        this.markerWidth = metrics.width + metrics.paddedWidth + metrics.offset;
+        this.markerHeight =
+            this.yAxis.toPixels(0) - this.yAxis.toPixels(calculatedBoxSize);
+
+        super.translate();
+    }
+}
+
+/* *
+ *
+ *  Class Prototype
+ *
+ * */
+
+interface PointAndFigureGroup {
+    x: number;
+    y: Array<number>;
+    upTrend: boolean;
+}
+
+interface PointAndFigureSeries {
+    takeOrdinalPosition: boolean;
+    pnfDataGroups: Array<PointAndFigureGroup>;
+    getColumnMetrics: typeof columnProto.getColumnMetrics;
+    markerWidth: number;
+    markerHeight: number;
+    calculatedBoxSize: number;
+}
+extend(PointAndFigureSeries.prototype, {
+    takeOrdinalPosition: true,
+    pnfDataGroups: [],
```

Comment:
this can be set on class with correct type, instead of using interface and extend.